### PR TITLE
Kanban: goals lazy pagination

### LINKED ISSUE #2913: Kanban: goals lazy pagination
support columns pagination

### PR DESCRIPTION
**Pull request by asabotovich**
## PR includes

- [x] Feature

## Related issues

Resolve #2913

## QA Instructions, Screenshots, Recordings
  

https://github.com/user-attachments/assets/5770aafc-88d0-4edb-bb54-35bd682d5f2d

### CODE REVIEW

File: src/components/Kanban/Kanban.tsx [hunk 1]
```diff
@@ -1,34 +1,119 @@
...
+    state: State;
+    projectId: string;
     filterPreset?: FilterById;
+    partnershipProject?: string[];
+    onLoadingStateChange?: onLoadingStateChange;
 }
 
-export const Kanban = ({ value, filterPreset }: KanbanProps) => {
-    const { data: states = [] } = trpc.state.all.useQuery();
+const calculateCommonLoadingState = (map: Map<unknown, LoadingState>): LoadingState => {
```
Comment (LamaEats) [inline src/components/Kanban/Kanban.tsx:31]: Why is it so difficult? Could you're move state calculations logic on column level? Cause that level has fetching the data

**Review by LamaEats — CHANGES_REQUESTED** [review]

File: src/utils/projectCollapsableItemInit.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
+import { QueryState } from '../hooks/useUrlFilterParams';
+
+import { SSRProps } from './declareSsrProps';
+
+export const filtersTakeCount = 5;
+
+export const projectCollapsableItemInit = async <T extends { id: string; partnerProjectIds?: string[] }>({
```
[the remark this answers is not in the document]
Comment (asabotovich) [reply src/utils/projectCollapsableItemInit.ts:7]: it used on server side

**Review by tsumo — APPROVED** [review]

File: src/components/Kanban/Kanban.tsx [hunk 3]
```diff
@@ -1,34 +1,119 @@
...
+
+    if (loadingStates.has('loading')) {
+        return 'loading';
+    }
+
+    if (loadingStates.size === 1 && loadingStates.has('empty')) {
+        return 'empty';
+    }
+
+    if (loadingStates.size === 1 && loadingStates.has('loaded')) {
```
Comment (9teen90nine) [inline src/components/Kanban/Kanban.tsx:42]: image

**Review by 9teen90nine — APPROVED** [review]

File: src/components/Kanban/Kanban.tsx [hunk 4]
```diff
@@ -42,66 +127,107 @@ export const Kanban = ({ value, filterPreset }: KanbanProps) => {
...
+        <>
+            <KanbanContainer>
+                {shownStates.map((state) => (
+                    <KanbanStateColumn
+                        key={state.id}
+                        projectId={id}
+                        state={state}
+                        filterPreset={filterPreset}
+                        partnershipProject={partnershipProject}
+                        onLoadingStateChange={(loadingState) =>
```
[the remark this answers is not in the document]
Comment (asabotovich) [reply src/components/Kanban/Kanban.tsx:214]: it has current item deps

**Pull request merged**